docs/guides/javascript/reactive/index.md

```diff
@@ -0,0 +1,1382 @@
+
+The method should return an array of objects containing:
+watch: the state changes to watch. For example `user.username:update`
+handler: the class method to handle the state change event. For example `this._refreshUsername`.
+
+<details>
+  <summary>View examples</summary>
+  <div>
+
+This is a real example of getWatchers from a course index module:
+
+```js
+getWatchers() {
+    return [
+        {watch: `cm:created`, handler: this._createCm},
+        {watch: `cm:deleted`, handler: this._deleteCm},
+        {watch: `section:created`, handler: this._createSection},
+        {watch: `section:deleted`, handler: this._deleteSection},
+    ];
+}
+```
+
+In case the component only watches changes in a specific element of a state collection, the function could be something like.
+
+```js
+getWatchers() {
+    const id = this.element.dataset.id;
+    return [
+        {watch: `cm[${id}]:created`, handler: this._createCm},
+        {watch: `cm[${id}]:deleted`, handler: this._deleteCm},
+        {watch: `section[${id}]:created`, handler: this._createSection},
+        {watch: `section[${id}]:deleted`, handler: this._deleteSection},
+    ];
+}
+```
+
+  </div>
+</details>
+
+:::note
+
+All the components code is designed to maintain the `this` value constant. This means that handlers don't need any kind of bind because the Reactive Instance will do it before registering the component. The same happens with event listeners.
+
+:::
+
+See the [Implement state watchers](#implementing-state-watchers) section for more information.
+
+  </TabItem>
+  <TabItem value="stateReady" label="stateReady">
+
+When the reactive instance has the initial state loaded. Is important to note that this method will be called even if the component is registered after the initial state is loaded. For all purposes, this method is the equivalent of a page ready for components.
+
+#### The `stateReady(state)` hook
+
+The Reactive Instance will call this hook on the registered components when the state data is initially loaded. This is one of the most important hooks in the Component lifecycle. At this moment the component will get its state and it can ensure that the Reactive Instance is ready and its watchers are working.
+
+Even if the component is registered once the initial state is loaded, the Reactive Instance will trigger that method when the Component is fully registered and the page is ready. For all purposes, this method is the equivalent of a page-ready event for components.
+
+The typical actions for a stateReady are:
+Add event listeners to capture user interactions.
+Refresh parts of the main element using the state data.
+Delegate logic to subcomponents.
+
+Go to the section [BaseComponent helpers](#basecomponent-helpers) to learn about some useful methods for simplifying the stateReady code.
```

Comment:
We tend to use "See the " rather than "Go to the", however since this is a _future_ topic it may be btter to say "The `BaseComponent` helpers section will discuss this in more detail." or similar.

Reply:
I like it. Fixed.